xs/src/libslic3r/ExPolygon.cpp

```diff
@@ -184,8 +184,104 @@ ExPolygon::simplify(double tolerance, ExPolygons* expolygons) const
     expolygons->insert(expolygons->end(), ep.begin(), ep.end());
 }
 
+/// remove point that are at SCALED_EPSILON * 2 distance.
 void
-ExPolygon::medial_axis(const ExPolygon &bounds, double max_width, double min_width, ThickPolylines* polylines) const
+remove_point_too_near(ThickPolyline* to_reduce) {
+        const int32_t smallest = SCALED_EPSILON * 2;
+        uint32_t id = 1;
+        while (id < to_reduce->points.size() - 2) {
+            uint32_t newdist = min(to_reduce->points[id].distance_to(to_reduce->points[id - 1])
+                , to_reduce->points[id].distance_to(to_reduce->points[id + 1]));
+            if (newdist < smallest) {
+                to_reduce->points.erase(to_reduce->points.begin() + id);
+                to_reduce->width.erase(to_reduce->width.begin() + id);
+            } else {
+                ++id;
+            }
+        }
+}
+
+/// add points  from pattern to to_modify at the same % of the length
+/// so not add if an other point is present at the correct position
+void
+add_point_same_percent(ThickPolyline* pattern, ThickPolyline* to_modify) {
+    const double to_modify_length = to_modify->length();
+    const double percent_epsilon = SCALED_EPSILON / to_modify_length;
+    const double pattern_length = pattern->length();
+
+    double percent_length = 0;
+    for (uint32_t idx_point = 1; idx_point < pattern->points.size() - 1; ++idx_point) {
+        percent_length += pattern->points[idx_point-1].distance_to(pattern->points[idx_point]) / pattern_length;
+        //find position 
+        uint32_t idx_other = 1;
+        double percent_length_other_before = 0;
+        double percent_length_other = 0;
+        while (idx_other < to_modify->points.size()) {
+            percent_length_other_before = percent_length_other;
+            percent_length_other += to_modify->points[idx_other-1].distance_to(to_modify->points[idx_other])
+                / to_modify_length;
+            if (percent_length_other > percent_length - percent_epsilon) {
+                //if higher (we have gone over it)
+                break;
+            }
+            ++idx_other;
+        }
+        if (percent_length_other > percent_length + percent_epsilon) {
+            //insert a new point before the position
+            double percent_dist = (percent_length - percent_length_other_before) / (percent_length_other - percent_length_other_before);
+            coordf_t new_width = to_modify->width[idx_other - 1] * (1 - percent_dist);
+            new_width += to_modify->width[idx_other] * (percent_dist);
+            Point new_point;
+            new_point.x = (coord_t)((double)(to_modify->points[idx_other - 1].x) * (1 - percent_dist));
```

Comment:
note: I forgot to remove the unnecessary (coord_t)((double)( )).
I don't see a cast problem.
Maybe i can pre-compute some things to remove some unnecessary operations here.

Reply:
the code truncates to both parts to integer before adding the (+=). Adding doubles and then truncating will slightly improve accuracy. But that is minot detail.

Function to blend/interpolate points (in point class) may be good idea - it will improve code readability

Reply:
`coord_t` is integer type